packages/grid/_modules_/grid/hooks/features/pagination/useGridPageSize.ts

```diff
@@ -19,15 +19,33 @@ import { gridDensityRowHeightSelector } from '../density';
  */
 export const useGridPageSize = (
   apiRef: GridApiRef,
-  props: Pick<GridComponentProps, 'pageSize' | 'onPageSizeChange' | 'autoPageSize'>,
+  props: Pick<
+    GridComponentProps,
+    'pageSize' | 'onPageSizeChange' | 'autoPageSize' | 'initialState'
+  >,
 ) => {
   const logger = useGridLogger(apiRef, 'useGridPageSize');
   const rowHeight = useGridSelector(apiRef, gridDensityRowHeightSelector);
 
-  useGridStateInit(apiRef, (state) => ({
-    ...state,
-    pagination: { pageSize: props.pageSize ?? (props.autoPageSize ? 0 : 100) },
-  }));
+  useGridStateInit(apiRef, (state) => {
+    let pageSize: number;
+    if (props.pageSize != null) {
+      pageSize = props.pageSize;
+    } else if (props.initialState?.pagination?.pageSize != null) {
+      pageSize = props.initialState.pagination.pageSize;
+    } else if (props.autoPageSize) {
+      pageSize = 0;
+    } else {
+      pageSize = 100;
+    }
+
+    return {
+      ...state,
+      pagination: {
+        pageSize,
```

Comment:
Should `state.pagination` be spread first? It doesn't cause a bug now because `useGridPage` comes after, but if it comes before this hook for some reason, then the initial page is lost.

Reply:
If the hook order is inverted, we would have other issues because `useGridPage` needs the page size to compute the `pageCount`.